### PR TITLE
closes #719

### LINKED ISSUE #719: Lower the handling fee from 3.5% to 3% and change the info message in the shopping cart accordingly
My Paypal account was approved for merchant rates based on J's volume. I checked and the rate currently being applies is 2.9%

We should pass this on to clients as it makes them more likely to keep using paypal rather than them switching to bank transfers.

Therefore change the info message to explain that the handling fee is 3% instead of 3.5% like we current have it and also change the actual handling fee itself.

We need to be ready to change it back at 24 hours notice because the rate of 2.9% will only be applied whilst we have a monthly volume of over 4k chf.

### PR DESCRIPTION
**Pull request by wszarmach**
Removed restriction on Invoice payments must be 7 days before first activity
Extracted PaymentOptions to own class
closes #719

### CODE REVIEW

File: schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/cart/PaymentOption.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            setPaymentType(PaymentTypeEnum.Invoice);
+            setDisabledReason(DisabledReason);
+            setEnabled(calculateEnabled(items));
+            setFee(InvoicePaymentProvider.INVOICE_FEE_FLAT.doubleValue());
+            setFeeReason(FeeReason);
+        }
+
+        private static String createFeeReason() {
+            String fee = formatNumber(InvoicePaymentProvider.INVOICE_FEE_FLAT, 2);
+            return "<nobr>For Electronic Invoice payments</nobr> there is a handling fee of <b><nobr>CHF " + fee + "</nobr></b>";
```
Comment (krico) [inline schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/cart/PaymentOption.java]: OMG, when we want to expand into germany and support EUR its going to be such a hassle! ;-)

File: schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/payment/InvoicePaymentProvider.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
  */
 public class InvoicePaymentProvider implements PaymentProvider<InvoicePayment> {
     public static final int INVOICE_EXPIRES_DAYS = 3;
-    public static final double INVOICE_FEE_FLAT = BigDecimal.ONE.doubleValue();
```
Comment (krico) [inline schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/payment/InvoicePaymentProvider.java]: yep, thought so...

File: schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/payment/PayPalPaymentProvider.java [hunk 3]
```diff
@@ -32,7 +32,7 @@
  * @since 05/02/15.
  */
 public class PayPalPaymentProvider implements PaymentProvider<PayPalPayment> {
-    public static final BigDecimal PAY_PAL_FEE_MULTIPLIER = new BigDecimal("0.035");
+    public static final BigDecimal PAY_PAL_FEE_MULTIPLIER = new BigDecimal("0.030");
```
Comment (krico) [inline schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/payment/PayPalPaymentProvider.java]: Cool!

File: schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/cart/PaymentOption.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        private static final Logger Log = LoggerFactory.getLogger(InvoicePaymentOption.class);
+        private static final long AllowedDays = 7;
+        private final static long InvoiceDaysBeforePaymentRequiredInMillis = 86400000 * AllowedDays;
+        private static final String DisabledReason = "For Invoice payments a minimum of " + AllowedDays + " days is required before the first activity";
+        private static final String FeeReason = createFeeReason();
+
+        public InvoicePaymentOption(List<ShoppingCart.Item> items) {
+            setPaymentType(PaymentTypeEnum.Invoice);
+            setDisabledReason(DisabledReason);
+            setEnabled(calculateEnabled(items));
```
Comment (krico) [inline schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/cart/PaymentOption.java]: I thought we said we would disable this for now...

Quoting yourself: Removed restriction on Invoice payments must be 7 days before first activity

File: schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/cart/PaymentOption.java (hunk 1 above)
Comment (wszarmach) [inline schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/cart/PaymentOption.java]: We could get rid of the text and pass the numeric values over to front end and do the translation on front end.

**Pull request merged**